File: ungs-docsys-ms/src/main/java/com/ungs/docsys/dtos/CertificationRequestDto.java
package com.ungs.docsys.dtos;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CertificationRequestDto {
    @NotBlank(message = "Name must not be blank")
    @Size(max = 150, message = "Name must not exceed 150 characters")
    private String name;
    @NotNull(message = "Issue date must not be null")
    private LocalDate issueDate;
    private LocalDate expirationDate;
    @NotBlank(message = "Certification URL must not be blank")
    @Size(max = 250, message = "Certification URL must not exceed 250 characters")
    private String certificationUrl;
}

File: ungs-docsys-bff/src/modules/resume-user/dtos/certification-request.dto.ts
import {
  IsString,
  IsOptional,
  IsDateString,
  MaxLength
} from 'class-validator';

export class CertificationRequestDto {
  @IsOptional()
  @IsString()
  @MaxLength(150, { message: 'Name must not exceed 150 characters' })
  name?: string;

  @IsOptional()
  @IsDateString({}, { message: 'Issue date must be a valid ISO 8601 date string' })
  issueDate?: string;

  @IsOptional()
  @IsDateString({}, { message: 'Expiration date must be a valid ISO 8601 date string' })
  expirationDate?: string;

  @IsOptional()
  @IsString()
  @MaxLength(250, { message: 'Certification URL must not exceed 250 characters' })
  certificationUrl?: string;
}